pkg/pmem-csi-driver/controllerserver-node.go

```diff
@@ -26,9 +26,10 @@ import (
 )
 
 type nodeVolume struct {
-	ID     string            `json:"id"`
-	Size   int64             `json:"size"`
-	Params map[string]string `json:"parameters"`
+	ID         string            `json:"id"`
+	DeviceMode string            `json:"deviceMode"`
+	Size       int64             `json:"size"`
+	Params     map[string]string `json:"parameters"`
```

Comment:
I would prefer a solution where "deviceMode" is a new parameter. We then don't allow setting it anywhere except in the `NodeVolumeOrigin` for now.

Should we ever decide to make it a parameter that can be set also in a storage class, then we don't need any special handling for it.

Reply:
> Should we ever decide to make it a parameter that can be set also in a storage class, then we don't need any special handling for it.

Making the driver's internal device mode feature a storage parameter sounds me not the right solution unless we plan to support creating volumes with chosen mode by admin/user.

Reply:
> Making the driver's internal device mode feature

That's an implementation detail of our current release. It might change.

> a storage parameter

Conceptually, it could be both a parameter for the volume ("give me a volume in direct mode") as well as a deployment option ("I want to run PMEM-CSI in direct mode"). Here we are dealing with volumes, so treating it as a parameter seems fine to me.

> unless we plan to support creating volumes with chosen mode by admin/user

We might, which is exactly the point I was trying to make.

Reply:
> Conceptually, it could be both a parameter for the volume ("give me a volume in direct mode")

I couldn't get why we need to expose this detail to the end-user. If the driver deployed in "lvm" mode and the user asks to create a volume in "direct" mode? or vice-versa. What is the benefit he gets?

Still, If this is what we are targetting, I feel like there is no point in choosing the device mode while deploying the driver.  So in my view, it should be either storage parameter(if we support provisioning of volumes in both modes) or deployment API option but not both.

Reply:
> I couldn't get why we need to expose this detail to the end-user.

If we as developers can't decide which mode is better, perhaps the admin cannot decide it either? It might depend on the application. Just a thought.

> I feel like there is no point in choosing the device mode while deploying the driver. So in my view, it should be either storage parameter(if we support provisioning of volumes in both modes) or deployment API option but not both.

Because we haven't decided about that, we now should pick an implementation that works for both.